packages/lib/src/ecdsa.ts

```diff
@@ -0,0 +1,70 @@
+const ECSignature = require("elliptic/lib/elliptic/ec/signature");
+const BN = require("bn.js");
+import { WeierstrassPoint, babyjubjub } from "./babyJubjub";
+import { Signature } from "./types";
+import { publicKeyFromString } from "./utils";
+
+export const privateKeyToPublicKey = (privKey: bigint): WeierstrassPoint => {
+  const pubKeyPoint = babyjubjub.ec.g.mul(privKey.toString(16));
+  return WeierstrassPoint.fromEllipticPoint(pubKeyPoint);
+};
+
+export const verifyEcdsaSignature = (
+  sig: Signature,
+  msgHash: bigint,
+  pubKey: WeierstrassPoint
+): boolean => {
+  const ecSignature = new ECSignature({
+    r: sig.r.toString(16),
+    s: sig.s.toString(16),
+  });
+  const pubKeyPoint = babyjubjub.ec.curve.point(
+    pubKey.x.toString(16),
+    pubKey.y.toString(16)
+  );
+  const ecPubKey = babyjubjub.ec.keyFromPublic(pubKeyPoint);
+
+  return babyjubjub.ec.verify(msgHash, ecSignature, ecPubKey);
+};
+
+export const recoverPubKeyIndexFromSignature = (
+  sig: Signature,
+  msgHash: bigint,
+  pubKeys: string[]
```

Comment:
Any reason this starts as string instead of just being stored as Edwards types?

Reply:
Decided to update all these signatures to use types within the package, per your suggestion